### PR TITLE
GetFeatureInfo components refactoring

### PR DESCRIPTION
**Pull request by annarieger**
* Refactors all GetFeatureInfo related components (Popup, Menu and Grid) to functional components
  * Replaced iconname based "close" and "download" icons for react-window toolbar by its analogs from `@ant-design/icons` library
  * Enhances typing
  * Replaced `width` property set on overlay by `maxWidth` (increased from 200 to 350px) to avoid cropped titles on long layer names
  * Some css fixes (padding/margin & friends)
  * Pass property `windowCollapsible` to menu component to control whether the GFI window should have collapse tool in the titlebar (default is true)

Please review @terrestris/devs

### CODE REVIEW

File: src/component/FeatureInfoGrid/FeatureInfoGrid.tsx [hunk 1]
```diff
@@ -24,144 +26,98 @@ import _upperFirst from 'lodash/upperFirst';
...
-      this.updateVectorLayer(selectedFeat);
+export const FeatureInfoGrid: React.FC<ComponentProps> = ({
+  downloadGridData,
+  features,
+  hoverVectorLayer,
+  isTimeLayer = false,
+  map,
+  onPaginationChange,
+  t
+}): JSX.Element => {
```
Comment (dnlkoch) [inline src/component/FeatureInfoGrid/FeatureInfoGrid.tsx:70]:
```suggestion
}): ReactElement => {
```

(This also might need an update of the imports)

File: src/component/FeatureInfoGrid/FeatureInfoGrid.tsx [hunk 2]
```diff
@@ -24,144 +26,98 @@ import _upperFirst from 'lodash/upperFirst';
...
+
+  const [selectedFeat, setSelectedFeat] = useState<OlFeature>(features[0]);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  useEffect(() => {
+    features[0].set('selectedFeat', true);
```
Comment (dnlkoch) [inline src/component/FeatureInfoGrid/FeatureInfoGrid.tsx:76]: Is `features[0]` always available?

File: src/component/FeatureInfoGrid/FeatureInfoGrid.tsx [hunk 3]
```diff
@@ -347,86 +293,62 @@ export class FeatureInfoGrid extends React.Component<FeatureInfoGridProps, Featu
...
         selectable={false}
         localeText={{
           noRowsToShow: t('FeatureInfoGrid.noDataFoundText')
         }}
         modules={[ClientSideRowModelModule, CsvExportModule]}
       />
     );
-  }
+  };
+
```
Comment (dnlkoch) [inline src/component/FeatureInfoGrid/FeatureInfoGrid.tsx:316]:
```suggestion
```

File: src/component/Popup/FeatureInfo/FeatureInfo.tsx [hunk 4]
```diff
@@ -85,148 +81,107 @@ interface FeatureInfoProps extends Partial<DefaultFeatureInfoProps> {
...
+  maxWidth = 300,
+  offset = [0, 0],
+  stopEvent = false,
+  insertFirst = true,
+  autoPan = true,
+  autoPanAnimationDuration = 250,
+  autoPanMargin = 20,
+  positioning,
+  position
+}): JSX.Element => {
```
Comment (dnlkoch) [inline src/component/Popup/FeatureInfo/FeatureInfo.tsx:128]: See above.

**Review by dnlkoch — APPROVED** [review]
Thanks!

File: src/component/Popup/FeatureInfo/FeatureInfo.tsx [hunk 5]
```diff
@@ -1,18 +1,16 @@
...
 import './FeatureInfo.css';
 
 interface DefaultFeatureInfoProps {
   /**
-   * The width of the popup.
+   * The max width of the popup.
    * @type {Number}
    */
-  width: number;
+  maxWidth?: number;
```
Comment (dnlkoch) [inline src/component/Popup/FeatureInfo/FeatureInfo.tsx:13]: Please note: This is a breaking change!

File: src/component/FeatureInfoGrid/FeatureInfoGrid.tsx (hunk 2 above)
Comment (annarieger) [reply src/component/FeatureInfoGrid/FeatureInfoGrid.tsx:76]: I think we are safe here. Features are coming from the context menu already grouped by feature types, for which the GetFeatureInfo request has returned at least one result. Empty results won't be listed there at all.

But an additional check surely doesn't hurt :)

File: src/component/Popup/FeatureInfo/FeatureInfo.tsx (hunk 5 above)
Comment (annarieger) [reply src/component/Popup/FeatureInfo/FeatureInfo.tsx:13]: I'll add a note about it into CHANGELOG.md later

File: src/component/Popup/FeatureInfo/FeatureInfo.tsx (hunk 4 above)
Comment (annarieger) [reply src/component/Popup/FeatureInfo/FeatureInfo.tsx]: S. e93029f

File: src/component/FeatureInfoGrid/FeatureInfoGrid.tsx (hunk 1 above)
Comment (annarieger) [reply src/component/FeatureInfoGrid/FeatureInfoGrid.tsx]: S. e93029f

File: src/component/FeatureInfoGrid/FeatureInfoGrid.tsx (hunk 2 above)
Comment (annarieger) [reply src/component/FeatureInfoGrid/FeatureInfoGrid.tsx:81]: Added in 34fb41f

**Pull request merged**